### PR TITLE
StableSwapModule audit fixes

### PR DESCRIPTION
**Pull request by vito-kovalione**
update StableSwapModule, fix existing calculation issues, add validations, update tests

### CODE REVIEW

File: contracts/main/stablecoin-core/StableSwapModule.sol [hunk 1]
```diff
@@ -178,22 +164,30 @@ contract StableSwapModule is PausableUpgradeable, ReentrancyGuardUpgradeable, IS
...
-            remainingDailySwapAmount= dailySwapLimit;
+            remainingDailySwapAmount = dailySwapLimit;
         }
-        require(remainingDailySwapAmount >= _amount,"_udpateAndCheckDailyLimit/daily-limit-exceeded");
+        require(remainingDailySwapAmount >= _amount, "_udpateAndCheckDailyLimit/daily-limit-exceeded");
         remainingDailySwapAmount -= _amount;
         emit LogRemainingDailySwapAmount(remainingDailySwapAmount);
     }
+
+    function _convertDecimals(
```
[the remark this answers is not in the document]
Comment (codinghistorian) [reply contracts/main/stablecoin-core/StableSwapModule.sol:186]: 2.1.5 withdrawFees only works for normal tokens in the StableSwapModule contract 

audit fix

**Review by codinghistorian — APPROVED** [review]
Stable Swap Module issues fixed

File: contracts/main/stablecoin-core/StableSwapModule.sol [hunk 2]
```diff
@@ -111,62 +98,61 @@ contract StableSwapModule is PausableUpgradeable, ReentrancyGuardUpgradeable, IS
...
+        uint256 stablecoinAmount = tokenAmount18 - fee;
+        require(tokenBalance[stablecoin] >= stablecoinAmount, "swapTokenToStablecoin/not-enough-stablecoin-balance");
+
+        _udpateAndCheckDailyLimit(tokenAmount18);
+
+        tokenBalance[stablecoin] -= stablecoinAmount;
+        tokenBalance[token] += _amount;
+        totalFeeBalance += fee;
+
+        token.safeTransferFrom(msg.sender, address(this), _amount);
```
Comment (codinghistorian) [inline contracts/main/stablecoin-core/StableSwapModule.sol:115]: _amount param is amount of token from which swap happens.
_amount param is amount of token that user pays

File: contracts/main/stablecoin-core/StableSwapModule.sol (hunk 1 above)
Comment (vito-kovalione) [inline contracts/main/stablecoin-core/StableSwapModule.sol:186]: this function was added for amount normalization, on order to have all calculations with 18 decimals

File: contracts/main/stablecoin-core/StableSwapModule.sol [hunk 3]
```diff
@@ -20,24 +20,21 @@ contract StableSwapModule is PausableUpgradeable, ReentrancyGuardUpgradeable, IS
     using SafeToken for address;
 
     IBookKeeper public bookKeeper;
-    IStablecoin public stablecoin;
+    address public stablecoin;
     address public token;
-    
-    uint256 internal to18ConversionFactor;
```
Comment (vito-kovalione) [inline contracts/main/stablecoin-core/StableSwapModule.sol:26]: we don't store conversion factor, but read decimals from token's contract instead

**Pull request merged**